### PR TITLE
Added a try/except for importing pam with logging an error and update…

### PR DESCRIPTION
**Pull request by blade2005**
…d README to note that when using pam you need to install pam module

```
2016-06-24 16:54:27 pypiserver._app INFO <LocalRequest: POST http://127.0.0.1:8092/>
2016-06-24 16:54:27 pypiserver.core ERROR PAM module not found. Please install pam module
2016-06-24 16:54:27 pypiserver._app INFO 200 OK
```

### CODE REVIEW

**Comment by blade2005** [comment]
On a successful upload with pam
```
2016-06-24 16:55:39 pypiserver._app INFO <LocalRequest: POST http://127.0.0.1:8092/>
2016-06-24 16:55:40 pypiserver._app INFO User u'blade2005' stored 'pypiserver-1.2.0b1.zip'.
2016-06-24 16:55:40 pypiserver._app INFO 200 OK
```

**Comment by mplanchard** [comment]
Test failure looks like it may have been non-deterministic. Re-running.

**Comment by blade2005** [comment]
LOL. I was so very confused when I was looking at the test and the results disappeared from my screen.

**Comment by mplanchard** [comment]
The beauty of distributed testing 🔮

**Comment by mplanchard** [comment]
Looks like it may be a real problem

**Comment by blade2005** [comment]
o.O how does this one fail testing where the last one didn't. Looks like bad request got returned when trying to upload. 

I tested it by hand and didn't see any issues.

```
# /usr/bin/pypi-server -r /packages -a '.' -P '.' -p 8092
```

```
removing 'pypiserver-1.2.0b1' (and everything under it)
running upload
Submitting dist/pypiserver-1.2.0b1.zip to http://127.0.0.1:8092/
Server response (200): OK
```

**Comment by mplanchard** [comment]
I’ll try running them again.

**Comment by mplanchard** [comment]
Hey, there we go.

**Comment by blade2005** [comment]
3rd time IS the charm then.

File: pypiserver/core.py [hunk 1]
```diff
@@ -92,8 +92,12 @@ def auth_by_htpasswd_file(htPsswdFile, username, password):
...
 
 def auth_by_pam(username, password):
-    import pam
-    return pam.authenticate(username, password)
+    try:
+        import pam
+        return pam.authenticate(username, password)
+    except ImportError as error:
+        log.error('PAM module not found. Please install pam module')
+        return False
```
Comment (ankostis) [inline pypiserver/core.py]: Sorry :-( 
We don't `try/catch` for missing dependencies - the original exception is quite explanatory to the user (in combination with the  documentation, which is the one needing improvement).

**Comment by mplanchard** [comment]
Huh. I didn't know that. Want to pull it out?

**Comment by ankostis** [comment]
Yes, lets keep the code under diet :-) thanks!

**Comment by mplanchard** [comment]
I'll have a PR up for that shortly

**Comment by mplanchard** [comment]
See #152

**Comment by ankostis** [comment]
@mplanchard please wait.
This PR is practically 2 lines of code + documentation:

    import pam
    ..., auther=pam.authenticate)

why don't we keep everything to the documentation, and leave the new code for a future release, when updating the `main()` flags?

**Comment by mplanchard** [comment]
You mean the `pam` import in general? I suppose we could pull it out entirely and push it to the next release. Seemed like a pretty easy include with docs, though. Whatever you think is best.

**Comment by ankostis** [comment]
Commented in #151 what i suggest.

**Pull request merged**